tools/integration_tests_runner.py

```diff
@@ -2,136 +2,130 @@
-            self.key_postfix += ' --user_resource_path="{0}"'.format(self.user_resource_path)
-        if self.data_path:
-            self.key_postfix += ' --data_path="{0}"'.format(self.data_path)
 
+def execute_file(test, keys):
+    spell = ["{test} {keys}".format(test=test, keys=keys)]
+    logging.debug(">> {spell}".format(spell=spell))
 
-    def run_tests(self):
-        intermediate_tests = []
-        for exec_file in self.runlist:
-            intermediate_tests.extend(map(lambda x: (exec_file, x), self.get_tests_from_exec_file(exec_file, "--list_tests")[0]))
+    process = subprocess.Popen(spell,
+                               stdout=subprocess.PIPE,
+                               stderr=subprocess.PIPE,
+                               shell=True
+                               )
 
-        shuffle(intermediate_tests)
-        for test in intermediate_tests:
-            self.tests.put(test)
+    out, err = process.communicate()
+    result = process.returncode
 
-        with open(self.output, "w") as self.file, open("start-finish.log", "w") as self.start_finish_log:
-            self.run_parallel_tests()
+    return filter(None, out.split("\n")), err, result
 
 
-    def run_parallel_tests(self):
-        threads = []
+def exec_test(a_tuple):
+    test_file, test, key_postfix = a_tuple
+    keys = '"--filter={test}"'.format(test=test)
 
-        for i in range(0, self.proc_count):
-            thread = Thread(target=self.exec_tests_in_queue)
-            thread.start()
-            threads.append(thread)
+    if test_file in TEMPFOLDER_TESTS:
+        tmpdir = tempfile.mkdtemp()
+        keys = '{old_key} "--data_path={tmpdir}"'.format(old_key=keys, tmpdir=tmpdir)
+        logging.debug("Temp dir: {tmpdir}".format(tmpdir=tmpdir))
+    else:
+        keys = "{old_key}{resource_path}".format(old_key=keys, resource_path=key_postfix)
+        logging.debug("Setting user_resource_path and data_path to {resource_path}".format(resource_path=key_postfix))
 
-        for thread in threads:
-            thread.join()
+    log_start_finish(test_file, keys, start=True)
+    out, err, result = execute_file(test_file, keys)
+    log_start_finish(test_file, keys, finish=True)
 
+    if test_file in TEMPFOLDER_TESTS:
+        try:
+            shutil.rmtree(tmpdir)
+        except:
+            logging.error("Failed to remove tempdir {tmpdir}".format(tmpdir=tmpdir))
 
-    def exec_tests_in_queue(self):
-        while not self.tests.empty():
-            try:
-                test_file, test = self.tests.get()
-                self.exec_test(test_file, test, clean_env=(test_file in TEMPFOLDER_TESTS))
+    tail = "\nEND: {file} | result: {res}\n\n".format(file=path.basename(test_file), res=result)
 
-            except:
-                logging.error(traceback.format_exc())
-                return
+    logging.getLogger(MAIN_LOGGER_NAME).info("BEGIN: {file}\n{message}{tail}".format(file=path.basename(test_file), message=str(err),  tail=tail))
```

Comment:
RTFM?

On 16 June 2016 at 11:41, therearesomewhocallmetim <notifications@github.com
> wrote:

> In tools/integration_tests_runner.py
> <https://github.com/mapsme/omim/pull/3575#discussion_r67307392>:
>
> >
> > -            except:
> > -                logging.error(traceback.format_exc())
> > -                return
> > +    logging.getLogger(MAIN_LOGGER_NAME).info("BEGIN: {file}\n{message}{tail}".format(file=path.basename(test_file), message=str(err),  tail=tail))
>
> А я могу в форматере логгера указать, что я буду передавать три
> переменные, и их нужно оформить таким-то образом? И потом смогу написать
> logger.info(file, message, result) и получить тот же самый результат?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/mapsme/omim/pull/3575/files/403f4e992cc2f19cc5406f315fcf267efb56ede4#r67307392>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACGjARW6UIBZlAfO64LEuTYgsqZbg0Jbks5qMQw9gaJpZM4I2caF>
> .
>